lldb/source/Core/FormatEntity.cpp

```diff
@@ -658,6 +658,33 @@ static char ConvertValueObjectStyleToChar(
   return '\0';
 }
 
+static bool DumpValueWithLLVMFormat(Stream &s, llvm::StringRef options,
+                                    ValueObject &target) {
+  std::string formatted;
+  std::string llvm_format = ("{0:" + options + "}").str();
```

Comment:
Is there any way we can make this string static, by switching over the supported options?
Or let me ask another way — what happens if options contained "}{1}" is this well-defined in llvm::formatv because it knows the template arguments and thus will not lead to corruption and crashes?
If the answer is yes, then this is okay.

Reply:
> by switching over the supported options?

There are many supported options, and they vary from type to type (int has different options than say strings). See [FormatProviders.h](https://llvm.org/doxygen/FormatProviders_8h_source.html) for the builtins.

It would be possible to exhaustively iterate all the options llvm includes, but I'm not sure it would be worth it. Switching over them seems fragile to changes made to the source of truth in llvm.

I will check/test what llvm does for invalid options.